include/svgpp/parser/grammar/color.hpp

```diff
@@ -140,6 +182,18 @@ class color_grammar: public qi::grammar<Iterator, typename ColorFactory::color_t
     return ColorFactory::create_from_percent(r, g, b);
   }
 
+#ifdef SVGPP_ACCEPT_RGBA_COLOR
+  static color_type absolute_rgba_components(unsigned char r, unsigned char g, unsigned char b, double a)
+  {
+    return ColorFactory::create(r, g, b, static_cast<unsigned char>(std::clamp(a, 0.0, 1.0) * 255 + 0.0049));
```

Comment:
Here I would suggest passing `a` as-is. Svgpp architecture is that parsers (`include/svgpp/parser/*`) just read SVG data with almost no conversions. And only after parsers user may place factories or adapters that makes data easier to handle.

Reply:
In that case, should I also remove the cast_alpha concept and pass without scaling/limiting the double to 255 bits? Otherwise, we'll have one create call that passes [0-255][0-255][0-255][0-255] and one that passes [0-255][0-255][0-255][0.0-1.0].

Reply:
You're right. Let's remove cast_alpha and just pass [0-1.0] double alpha around. Casting alpha to 0-255 if needed will either be done in client code or in some new adapter/factory in svgpp.